cpp/bench/prims/neighbors/knn.cuh

```diff
@@ -156,6 +158,46 @@ struct ivf_flat_knn {
   }
 };
 
+template <typename ValT, typename IdxT>
+struct cagra_knn {
+  using dist_t = float;
+
+  std::optional<const raft::neighbors::experimental::cagra::index<ValT, IdxT>> index;
+  raft::neighbors::experimental::cagra::index_params index_params;
+  raft::neighbors::experimental::cagra::search_params search_params;
+  params ps;
+
+  cagra_knn(const raft::device_resources& handle, const params& ps, const ValT* data) : ps(ps)
+  {
+    // index_params.intermediate_graph_degree = 128;
+    // index_params.graph_degree = 64;
+    index_params.metric = raft::distance::DistanceType::L2Expanded;
+
+    auto dataset_view =
+      raft::make_device_matrix_view<const ValT, IdxT>(data, ps.n_samples, ps.n_dims);
+    index.emplace(raft::neighbors::experimental::cagra::build(handle, index_params, dataset_view));
+  }
+
+  void search(const raft::device_resources& handle,
+              const ValT* search_items,
+              dist_t* out_dists,
+              IdxT* out_idxs)
+  {
+    // search_params.algo        = ps.algo;
```

Comment:
Is this being kept around for any reason?

Reply:
removed. Currently we have auto selection for algo, we can update that if we see a need for it.